utils/preprocess_utils.py

```diff
@@ -22,34 +22,38 @@ def __init__(self, path):
             Metadata column names to be kept in the retrieved dataframe (optional)
         """
         path = pathlib.Path(path)
-                     
+
         # If the file isn't found in the path above then raise an error.
         if not path.is_file():
             raise FileNotFoundError(f"File '{path}' does not exist")
-            
-        if 'parquet' in path.name:
+
+        if "parquet" in path.name:
```

Comment:
Consider the use of [`pathlib.Path.suffix`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix) for this and other lines.

Reply:
I'm not sure if the parquet files can still be read after using gzip. I think the `pathlib.Path.suffix` only considers the text after the last `.`.